### PR TITLE
Calendroid: Calendar adapter

### LINKED ISSUE #4285: Create the CalendarEventsAdapter
This is the adapter that will be bound to the calendar events.
There should be a method to update the current time entries.
It won't hurt to implement DiffUtil for this adapter.

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
This is the calendar adapter for the calendar view o/

### Relationships
Closes #4285 

## Why do we want this?
We've gotta display the items in the calendar o/

## How is it done?
It uses the `CalendarLayoutCalculator` to calculate where the events should be positioned. That's it :)

### Why not another way?
Feel free to propose something else. We are using this because we went with a custom layout manager instead of a custom view for the calendar.

### Side effects
There shouldn't be any.

## Review hints
None.

## :squid: Permissions
🦑 after WIP.

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 using System.Reactive.Subjects;
 using System.Reactive.Threading.Tasks;
 using System.Threading.Tasks;
+using MvvmCross.Logging;
```
Comment (juanlaube) [inline Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs]: 👀

File: Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs [hunk 2]
```diff
@@ -75,6 +80,8 @@ public sealed class CalendarViewModel : MvxViewModel
 
         public ObservableGroupedOrderedCollection<CalendarItem> CalendarItems { get; }
 
+        public bool HasCalendarsLinkedLastValue => hasCalendarsLinked;
```
Comment (juanlaube) [inline Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs]: Make this an observable

File: Toggl.Giskard/Adapters/Calendar/CalendarAdapter.cs [hunk 3]
```diff
@@ -1,56 +1,100 @@
 using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
```
Comment (juanlaube) [inline Toggl.Giskard/Adapters/Calendar/CalendarAdapter.cs]: Remove unneeded usings in this file

File: Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs (hunk 2 above)
Comment (daividssilverio) [reply Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs]: It already is though, I created that so I can set the initial value for the adapter in the `CalendarFragment`:

```
calendarAdapter.UpdateItems(ViewModel.CalendarItems, ViewModel.HasCalendarsLinkedLastValue);`
```
Any suggestions?

Comment (daividssilverio) [reply Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs]: I've created `public IObservable<bool> HasCalendarsLinked { get; }` but I need some initial value immediately.

Comment (daividssilverio) [reply Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs]: I've removed the `HasCalendarsLinkedLastValue` and replaced it another strategy to detect updates. Take a look.

File: Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs [hunk 4]
```diff
@@ -213,6 +210,8 @@ public async override Task Initialize()
                 .Merge(dayChangedObservable)
                 .Merge(selectedCalendarsChangedObservable)
                 .Merge(appResumedFromBackgroundObservable)
+                .ObserveOn(schedulerProvider.BackgroundScheduler)
```
[the remark this answers is not in the document]
Comment (daividssilverio) [reply Toggl.Foundation.MvvmCross/ViewModels/Calendar/CalendarViewModel.cs]: Tested, and you are right o/

**Pull request merged**